_studio/shared/src/mfx_umc_alloc_wrapper.cpp

```diff
@@ -1104,14 +1105,14 @@ mfxStatus   mfx_UMC_FrameAllocator_D3D::PrepareToOutput(mfxFrameSurface1 *surfac
             surface.Data.Y = 0;
             surface.Data.MemId = idx;
             //Performance issue. We need to unlock mutex to let decoding thread run async.
-            guard.Unlock();
+            guard.unlock();
```

Comment:
Again, not much sense in manual unlocks of ```recursive_mutex```

Reply:
@mgonchar,

But it still makes sense - when guard is locked only once it enables copying w/o lock.

Reply:
@dmitry-gurulev yes, but it doesn't give any hint on why it is required. So it might be locked (if some other routine of this thread captured this mutex) or unlocked at this point. If copying works only when mutex is unlocked - we have a problem, if copying doesn't care about state of the mutex then I want to understand how possibility of that mutex to be unlocked helps to perform copy

Reply:
@mgonchar,

AFAIR, copying itself doesn't care about state, unlocking just allow other threads do their work not waiting when copying finishes.

Reply:
@dmitry-gurulev so this is kind of optimization that might happen if ```mutex``` ownership was exclusive. Well, better to look at this architecture closer, because such places looks suspicious